hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -359,7 +370,7 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 			removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolValidAMIConditionType)
 		} else {
 			// TODO: Should the region be included in the NodePool platform information?
-			ami, err = defaultNodePoolAMI(hcluster.Spec.Platform.AWS.Region, releaseImage)
+			ami, err = defaultNodePoolAMI(hcluster.Spec.Platform.AWS.Region, nodePool.Spec.Arch, releaseImage)
```

Comment:
we need to validate we are in AWS and set a condition signalling this has no effect for other platforms

Reply:
Ok I think I understand and added a setStatusCondition for this.